shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -349,555 +349,640 @@ class SpellCheckerHolder final : public content::RenderFrameObserver {
+    } else if (pref_name == options::kOffscreen) {
+      return gin::ConvertToV8(isolate, prefs.offscreen);
+    } else if (pref_name == options::kPreloadScript) {
+      return gin::ConvertToV8(isolate, prefs.preload.value());
+    } else if (pref_name == options::kNativeWindowOpen) {
+      return gin::ConvertToV8(isolate, prefs.native_window_open);
+    } else if (pref_name == options::kNodeIntegration) {
+      return gin::ConvertToV8(isolate, prefs.node_integration);
+    } else if (pref_name == options::kNodeIntegrationInWorker) {
+      return gin::ConvertToV8(isolate, prefs.node_integration_in_worker);
+    } else if (pref_name == options::kEnableNodeLeakageInRenderers) {
+      // NOTE: enableNodeLeakageInRenderers is internal-only.
+      return gin::ConvertToV8(isolate, prefs.node_leakage_in_renderers);
+    } else if (pref_name == options::kNodeIntegrationInSubFrames) {
+      return gin::ConvertToV8(isolate, true);
+#if BUILDFLAG(ENABLE_BUILTIN_SPELLCHECKER)
+    } else if (pref_name == options::kSpellcheck) {
+      return gin::ConvertToV8(isolate, prefs.enable_spellcheck);
+#endif
+    } else if (pref_name == options::kPlugins) {
+      return gin::ConvertToV8(isolate, prefs.enable_plugins);
+    } else if (pref_name == options::kEnableWebSQL) {
+      return gin::ConvertToV8(isolate, prefs.enable_websql);
+    } else if (pref_name == options::kWebviewTag) {
+      return gin::ConvertToV8(isolate, prefs.webview_tag);
+    }
+    return v8::Null(isolate);
   }
-  return std::u16string();
-}
 
-void RemoveInsertedCSS(gin_helper::ErrorThrower thrower,
-                       v8::Local<v8::Value> window,
-                       const std::u16string& key) {
-  auto* render_frame = GetRenderFrame(window);
-  if (!render_frame) {
-    thrower.ThrowError(
-        "Render frame was torn down before webFrame.removeInsertedCSS could be "
-        "executed");
-    return;
+  void SetVisualZoomLevelLimits(gin_helper::ErrorThrower thrower,
+                                double min_level,
+                                double max_level) {
+    content::RenderFrame* render_frame;
+    if (!MaybeGetRenderFrame(thrower, "setVisualZoomLevelLimits",
+                             &render_frame))
+      return;
+
+    blink::WebFrame* web_frame = render_frame->GetWebFrame();
+    web_frame->View()->SetDefaultPageScaleLimits(min_level, max_level);
   }
 
-  blink::WebFrame* web_frame = render_frame->GetWebFrame();
-  if (web_frame->IsWebLocalFrame()) {
-    web_frame->ToWebLocalFrame()->GetDocument().RemoveInsertedStyleSheet(
-        blink::WebString::FromUTF16(key));
+  void AllowGuestViewElementDefinition(gin_helper::ErrorThrower thrower,
+                                       v8::Local<v8::Object> context,
+                                       v8::Local<v8::Function> register_cb) {
+    v8::HandleScope handle_scope(thrower.isolate());
+    v8::Context::Scope context_scope(context->CreationContext());
```

Comment:
Why does this method have a handle / context scope when none of the others do? It doesn't look like it's ever called from outside of V8.

Also, is `context` still needed here?

Reply:
I don't know, this method was copy-pasted.  Happy to take a look at some point but I don't want to change actual implementations here.